docs/libraries/go.md

```diff
@@ -127,10 +129,10 @@ func main() {
 	ctx := context.TODO()
 	zctx := zed.NewContext()
 	var lake api.Interface
-	if api.IsLakeService(uri) {
-		lake, err = api.OpenRemoteLake(ctx, uri.String())
+	if api.IsLakeService(uri.String()) {
+		lake, err = api.OpenLake(ctx, uri.String())
 	} else {
-		lake, err = api.OpenLocalLake(ctx, uri)
+		lake, err = api.OpenLocalLake(ctx, uri.String())
```

Comment:
While I was able to get it to compile these changes, upon first run attempt, it complained of a bunch of missing dependencies:

```
$ go run . ./scratch
../.go/pkg/mod/github.com/brimdata/zed@v1.1.0/pkg/units/bytes.go:6:2: missing go.sum entry for module providing package github.com/alecthomas/units (imported by github.com/brimdata/zed/pkg/units); to add:
	go get github.com/brimdata/zed/pkg/units@v1.1.0
../.go/pkg/mod/github.com/brimdata/zed@v1.1.0/runtime/expr/cast.go:9:2: missing go.sum entry for module providing package github.com/araddon/dateparse (imported by github.com/brimdata/zed/runtime/expr); to add:
	go get github.com/brimdata/zed/runtime/expr@v1.1.0
../.go/pkg/mod/github.com/brimdata/zed@v1.1.0/pkg/s3io/reader.go:10:2: missing go.sum entry for module providing package github.com/aws/aws-sdk-go/aws (imported by github.com/brimdata/zed/pkg/s3io); to add:
	go get github.com/brimdata/zed/pkg/s3io@v1.1.0
../.go/pkg/mod/github.com/brimdata/zed@v1.1.0/pkg/storage/s3.go:10:2: missing go.sum entry for module providing package github.com/aws/aws-sdk-go/aws/awserr (imported by github.com/brimdata/zed/pkg/storage); to add:
	go get github.com/brimdata/zed/pkg/storage@v1.1.0
../.go/pkg/mod/github.com/brimdata/zed@v1.1.0/pkg/s3io/s3io.go:17:2: missing go.sum entry for module providing package github.com/aws/aws-sdk-go/aws/session (imported by github.com/brimdata/zed/pkg/s3io); to add:
	go get github.com/brimdata/zed/pkg/s3io@v1.1.0
../.go/pkg/mod/github.com/brimdata/zed@v1.1.0/pkg/s3io/reader.go:11:2: missing go.sum entry for module providing package github.com/aws/aws-sdk-go/service/s3 (imported by github.com/brimdata/zed/pkg/s3io); to add:
	go get github.com/brimdata/zed/pkg/s3io@v1.1.0
../.go/pkg/mod/github.com/brimdata/zed@v1.1.0/pkg/storage/s3.go:11:2: missing go.sum entry for module providing package github.com/aws/aws-sdk-go/service/s3/s3iface (imported by github.com/brimdata/zed/pkg/storage); to add:
	go get github.com/brimdata/zed/pkg/storage@v1.1.0
../.go/pkg/mod/github.com/brimdata/zed@v1.1.0/pkg/s3io/s3io.go:20:2: missing go.sum entry for module providing package github.com/aws/aws-sdk-go/service/s3/s3manager (imported by github.com/brimdata/zed/pkg/s3io); to add:
	go get github.com/brimdata/zed/pkg/s3io@v1.1.0
../.go/pkg/mod/github.com/brimdata/zed@v1.1.0/runtime/expr/agg/dcount.go:4:2: missing go.sum entry for module providing package github.com/axiomhq/hyperloglog (imported by github.com/brimdata/zed/runtime/expr/agg); to add:
	go get github.com/brimdata/zed/runtime/expr/agg@v1.1.0
../.go/pkg/mod/github.com/brimdata/zed@v1.1.0/zio/parquetio/reader.go:8:2: missing go.sum entry for module providing package github.com/fraugster/parquet-go (imported by github.com/brimdata/zed/zio/parquetio); to add:
	go get github.com/brimdata/zed/zio/parquetio@v1.1.0
../.go/pkg/mod/github.com/brimdata/zed@v1.1.0/zio/parquetio/schemadefinition.go:9:2: missing go.sum entry for module providing package github.com/fraugster/parquet-go/parquet (imported by github.com/brimdata/zed/zio/parquetio); to add:
	go get github.com/brimdata/zed/zio/parquetio@v1.1.0
../.go/pkg/mod/github.com/brimdata/zed@v1.1.0/zio/parquetio/schemadefinition.go:10:2: missing go.sum entry for module providing package github.com/fraugster/parquet-go/parquetschema (imported by github.com/brimdata/zed/zio/parquetio); to add:
	go get github.com/brimdata/zed/zio/parquetio@v1.1.0
../.go/pkg/mod/github.com/brimdata/zed@v1.1.0/lake/commits/store.go:17:2: missing go.sum entry for module providing package github.com/hashicorp/golang-lru (imported by github.com/brimdata/zed/lake); to add:
	go get github.com/brimdata/zed/lake@v1.1.0
../.go/pkg/mod/github.com/brimdata/zed@v1.1.0/pkg/charm/help.go:9:2: missing go.sum entry for module providing package github.com/kr/text (imported by github.com/brimdata/zed/pkg/charm); to add:
	go get github.com/brimdata/zed/pkg/charm@v1.1.0
../.go/pkg/mod/github.com/brimdata/zed@v1.1.0/zio/zngio/frame.go:6:2: missing go.sum entry for module providing package github.com/pierrec/lz4/v4 (imported by github.com/brimdata/zed/zio/zngio); to add:
	go get github.com/brimdata/zed/zio/zngio@v1.1.0
../.go/pkg/mod/github.com/brimdata/zed@v1.1.0/compiler/ast/dag/operator.go:13:2: missing go.sum entry for module providing package github.com/segmentio/ksuid (imported by github.com/brimdata/zed/lake/api); to add:
	go get github.com/brimdata/zed/lake/api@v1.1.0
../.go/pkg/mod/github.com/brimdata/zed@v1.1.0/lake/index/combiner.go:9:2: missing go.sum entry for module providing package go.uber.org/multierr (imported by github.com/brimdata/zed/lake/index); to add:
	go get github.com/brimdata/zed/lake/index@v1.1.0
../.go/pkg/mod/github.com/brimdata/zed@v1.1.0/runtime/op/op.go:12:2: missing go.sum entry for module providing package go.uber.org/zap (imported by github.com/brimdata/zed/runtime); to add:
	go get github.com/brimdata/zed/runtime@v1.1.0
../.go/pkg/mod/github.com/brimdata/zed@v1.1.0/lake/scheduler.go:19:2: missing go.sum entry for module providing package golang.org/x/sync/errgroup (imported by github.com/brimdata/zed/lake); to add:
	go get github.com/brimdata/zed/lake@v1.1.0
../.go/pkg/mod/github.com/brimdata/zed@v1.1.0/lake/index/filter.go:19:2: missing go.sum entry for module providing package golang.org/x/sync/semaphore (imported by github.com/brimdata/zed/lake/index); to add:
	go get github.com/brimdata/zed/lake/index@v1.1.0
../.go/pkg/mod/github.com/brimdata/zed@v1.1.0/pkg/terminal/terminal.go:6:2: missing go.sum entry for module providing package golang.org/x/term (imported by github.com/brimdata/zed/pkg/terminal); to add:
	go get github.com/brimdata/zed/pkg/terminal@v1.1.0
```

I was able to make it work by grepping out and executing every one of those `go get` suggestions. However, I have to assume there's a better way to get all those dependencies, so rather than paste all those lines into the doc I figured I'd ask here first.

Reply:
@nwt: Right you are. This time around it gave me the much friendlier:

```
$ go run . ./scratch
go: updates to go.mod needed; to update it:
	go mod tidy
```

And indeed once I ran `go mod tidy`, it downloaded the dependencies and then I could run successfully.

Assuming you accept my approach from the earlier in the doc that added `go mod tidy`, are you ok with leaving out explicit mention of it here again to keep the text simple? Since it was already mentioned once and to `go` binary is speaking clearly, it seems that would be ok.